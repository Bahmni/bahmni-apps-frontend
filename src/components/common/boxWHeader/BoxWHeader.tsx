import React, { ReactNode } from 'react';
import { Column, Grid } from '@carbon/react';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';
import * as styles from './styles/BoxWHeader.module.scss';

/**
 * BoxWHeader Props
 * @interface BoxWHeaderProps
 */
export interface BoxWHeaderProps {
  /**
   * The title to display in the header.
   * If the title is a translation key, it will be automatically translated.
   */
  title: string;

  /**
   * The content to display in the box body
   */
  children: ReactNode;

  /**
   * Optional CSS class name to apply to the container
   */
  className?: string;

  /**
   * Optional data-testid attribute for testing
   * @default 'box-w-title'
   */
  dataTestId?: string;
  /**
   * Optional ARIA label for the section
   * If not provided, the title will be used
   */
  ariaLabel?: string;
}

/**
 * BoxWHeader component
 *
 * A container with a header title and content area.
 * Uses Carbon Design Grid for responsive layout.
 */
const BoxWHeader: React.FC<BoxWHeaderProps> = ({
  title,
  children,
  className,
  dataTestId = 'box-w-title',
  ariaLabel,
}) => {
  const { t } = useTranslation();

  // Translate title if it's a translation key
  const translatedTitle = t(title);

  return (
    <Grid
      narrow
      fullWidth
      className={classNames(styles.box, className)}
      data-testid={dataTestId}
      aria-label={ariaLabel || translatedTitle}
      role="region"
    >
      <Column sm={4} md={8} lg={16} xlg={16} className={styles.header}>
        {translatedTitle}
      </Column>
      <Column sm={4} md={8} lg={16} xlg={16}>
        {children}
      </Column>
    </Grid>
  );
};

export default BoxWHeader;
