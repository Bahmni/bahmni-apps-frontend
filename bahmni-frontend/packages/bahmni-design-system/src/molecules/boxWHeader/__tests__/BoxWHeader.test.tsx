import { render, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { BoxWHeader } from '../BoxWHeader';

expect.extend(toHaveNoViolations);
// Mock the styles to prevent issues with CSS modules in tests
jest.mock('../styles/BoxWHeader.module.scss', () => ({
  box: 'box-class',
  header: 'header-class',
}));

describe('BoxWHeader', () => {
  describe('Happy Paths', () => {
    it('should render correctly with required props', () => {
      // Arrange
      const title = 'Test Title';
      const childText = 'Test Children';

      // Act
      render(
        <BoxWHeader title={title}>
          <div>{childText}</div>
        </BoxWHeader>,
      );

      // Assert
      expect(screen.getByText(title)).toBeInTheDocument();
      expect(screen.getByText(childText)).toBeInTheDocument();
      expect(screen.getByTestId('box-w-title')).toBeInTheDocument();
    });

    it('should render with custom data-testid when provided', () => {
      // Arrange
      const customTestId = 'custom-test-id';

      // Act
      render(
        <BoxWHeader title="Title" dataTestId={customTestId}>
          <div>Content</div>
        </BoxWHeader>,
      );

      // Assert
      expect(screen.getByTestId(customTestId)).toBeInTheDocument();
    });

    it('should apply custom className when provided', () => {
      // Arrange
      const customClass = 'custom-class';

      // Act
      const { container } = render(
        <BoxWHeader title="Title" className={customClass}>
          <div>Content</div>
        </BoxWHeader>,
      );

      // Assert
      const gridElement = container.firstChild;
      expect(gridElement).toHaveClass('box-class');
      expect(gridElement).toHaveClass(customClass);
    });

    it('should match snapshot', () => {
      // Arrange & Act
      const { container } = render(
        <BoxWHeader title="Snapshot Test">
          <div>Snapshot Content</div>
        </BoxWHeader>,
      );

      // Assert
      expect(container).toMatchSnapshot();
    });
  });

  describe('Accessibility', () => {
    it('should not have accessibility violations', async () => {
      // Arrange
      const { container } = render(
        <BoxWHeader title="Accessibility Test">
          <div>Test content for accessibility</div>
        </BoxWHeader>,
      );

      // Act & Assert
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
});
