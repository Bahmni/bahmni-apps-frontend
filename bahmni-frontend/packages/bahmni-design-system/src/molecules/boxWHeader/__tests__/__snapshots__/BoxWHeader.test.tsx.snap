// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxWHeader Happy Paths should match snapshot 1`] = `
<div>
  <div
    aria-label="Snapshot Test"
    class="box-class cds--css-grid cds--css-grid--narrow cds--css-grid--full-width"
    data-testid="box-w-title"
    role="region"
  >
    <div
      class="header-class cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-16 cds--xlg:col-span-16 cds--css-grid-column"
    >
      Snapshot Test
    </div>
    <div
      class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-16 cds--xlg:col-span-16 cds--css-grid-column"
    >
      <div>
        Snapshot Content
      </div>
    </div>
  </div>
</div>
`;
